packages/base-controller/src/ControllerMessenger.ts

```diff
@@ -430,12 +430,12 @@ export class ControllerMessenger<
    */
   getRestricted<
     Namespace extends string,
-    AllowedAction extends NotNamespacedBy<Namespace, Action['type']>,
-    AllowedEvent extends NotNamespacedBy<Namespace, Event['type']>,
+    AllowedAction extends NotNamespacedBy<Namespace, Action['type']> = never,
+    AllowedEvent extends NotNamespacedBy<Namespace, Event['type']> = never,
   >({
     name,
-    allowedActions,
-    allowedEvents,
+    allowedActions = [],
```

Comment:
It looks like the drop in coverage is caused by this line. This makes the `?? null` branch in the `RestrictedControllerMessenger` unreachable.

Perhaps we could remove that branch, and change the `allowedActions` and `allowedEvents` parameters to be required in the `RestrictedControllerMessenger`  constructor? That would eliminate the unreachable branch and restore full coverage.